extensions/common/shared.hpp

```diff
@@ -20,6 +20,12 @@
 #include "ace_version.hpp"
 #include <algorithm>
 #include <cctype> //std::isspace
+#include <cstring>
+#include <istream>
+
+#ifndef _WIN32
+#define strncpy_s(dest, size, src, ignored) strncpy(dest, src, size)
```

Comment:
I would prefer if we just disable MSVC "safe" warnings and use proper C++ functions everywhere.

Reply:
strncpy_s is C.
I guess the proper C++ std::strncpy function should work crossplatform.

Reply:
Does std::strncpy throw warnings in MSVC too or are they okay with that one?

Reply:
I think it might throw warnings as well, but I am not sure.

Reply:
It might throw unchecked iterators warning/critical error. But as it takes char* and not iterators it really shouldn't.

Reply:
On Intercept I use `std::copy_n(src, len, dest)` which doesn't automatically add the ending null char which I think strncpy does. I just set it manually. like `dest[ignored] = 0;`

Reply:
Will disable MSVC "unsafe" warnings for all, but in separate PR.

Reply:
-> #7171